tests/test_all_simple_paths.py

```diff
@@ -0,0 +1,347 @@
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import unittest
+
+import retworkx
+
+
+class TestDAGAllSimplePaths(unittest.TestCase):
+    def test_all_simple_paths(self):
+        dag = retworkx.PyDAG()
+        for i in range(6):
+            dag.add_node(i)
+        edges = [
+            (0, 1),
+            (0, 2),
+            (0, 3),
+            (1, 2),
+            (1, 3),
+            (2, 3),
+            (2, 4),
+            (3, 2),
```

Comment:
2->3 and 3->2 makes a cycle, so this shouldn't be a DAG... I assume DAG checking is off by default?

Reply:
Yeah, see my earlier comment. By default `PyDAG` isn't actually a DAG. This was done for performance reasons in https://github.com/Qiskit/retworkx/pull/28 so unless we set it to true manually this is just a directed graph